xdsl/dialects/memref_stream.py

```diff
@@ -310,6 +319,12 @@ class GenericOp(IRDLOperation):
     pattern defines the order in which the elements of the input buffers will be written
     to.
     """
+    inits = prop_def(ArrayAttr[AnyFloatAttr | AnyIntegerAttr | UnitAttr])
+    """
+    Initial values for outputs. If `NoneAttr`, then the value is read from the output
+    buffer. Otherwise, the value is created at runtime with an `arith.constant` operation
+    during lowering. The inits may be set only for the imperfectly nested form.
```

Comment:
The comment mentions NoneAttr but the prop uses UnitAttr!

I skimmed your unit PR and I'm not sure I get it; why not NoneAttr? The name sounds more like the use here?

Reply:
Our printing and parsing of it is somewhat broken, NoneAttr doesn't roundtrip in the IR, as `#none` is interpreted as an alias, and `none` is NoneType, which seems like the wrong thing to use. This is one bit I was particularly hoping for feedback/ideas of how to do better. I could add custom parsing and printing for this, and spell the missing values as `_` or add them somewhere else in the custom syntax, also. Recommendations welcome.